### ISSUE TITLE
noUnusedLocals incorrectly flags function arguments

### ISSUE THREAD

**Issue by dcbrwn** [microsoft/TypeScript#47906]
### 🔎 Search Terms

- noUnusedLocals
- noUnusedLocals works incorrectly
- TS6133
- TS6133 false-positive
- property is declared but never used

### 🕗 Version & Regression Information

- This is the behavior in every version between 4.5.4 and 3.3.3

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?noUnusedParameters=true&noUnusedLocals=true#code/C4TwDgpgBAglC8UAUAzA9mgXFAdgVwFsAjCAJwBooiBDAL2wGdhSBLHAcwEoEA+KANzQsAJgG4AUOJR4cAY2As0OKtVJJO2OAG9xUKKQjA8pZagzZ8xMpRr0oTVhw0Chw3lB169spQzQAbCAA6fzR2JFtOCT0AX3E4oA
)

### 💻 Code

```ts
type A = (foo: number, baz: string) => void;

function bar(): A {
  return (foo: number, baz: string): void => {
    console.log(baz);
  }
}
```

### 🙁 Actual behavior

`tsc` with noUnusedLocals turned on, complains about `foo` in the returned arrow function. But we can't make it used (because this changes internal logic of the function), nor can we change the type of returned function (because this changes public interface). `noUnusedLocals` is useful and is a part of our common config, so turning it off is not an option. The only solution yet is to `// @ts-ignore`.

### 🙂 Expected behavior

No errors reported.

**Comment by MartinJohns** [comment]
> The only solution yet is to `// @ts-ignore`.

That's not correct. You can prefix the variable name with `_` instead.

**Comment by dcbrwn** [comment]
Ok, thanks! I'll take a look

**Comment by MartinJohns** [comment]
> In our case there is no linter/code-style rule that prevents this naming style, but i've seen some in the wild. So prefixing with underscore may not be a solution for everyone.

You should configure the tools to your need, not blindly follow whatever suggestion and rule they make.

Another solution, although less nice, would be to simply use the argument in a no-op:

```
void foo;
```

**Comment by dcbrwn** [comment]
> You should configure the tools to your need, not blindly follow whatever suggestion and rule they make.

In practice, changing said configuration is not always possible. Some people stuck with what they have, because CTOs or cometees decided something for them,
